test/boost/error_injection_test.cc

```diff
@@ -6,6 +6,7 @@
  * SPDX-License-Identifier: AGPL-3.0-or-later
  */
 
+#include <seastar/testing/on_internal_error.hh>
```

Comment:
re commit message:
"in cases with concurrency problems"

I know concurrency usually means problems :) But I would simply say "in presence of concurrency" - these are just natural situations which may arise in a concurrent environment, not "problems".